File: chat-uikit/src/main/java/io/agora/chat/uikit/widget/chatrow/EaseChatRowBigExpression.java
package io.agora.chat.uikit.widget.chatrow;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.core.content.ContextCompat;

import com.bumptech.glide.Glide;
import com.bumptech.glide.request.RequestOptions;

import io.agora.chat.ChatMessage;
import io.agora.chat.uikit.EaseUIKit;
import io.agora.chat.uikit.R;
import io.agora.chat.uikit.chat.model.EaseChatItemStyleHelper;
import io.agora.chat.uikit.constants.EaseConstant;
import io.agora.chat.uikit.models.EaseEmojicon;
import io.agora.chat.uikit.utils.EaseUtils;

/**
 * big emoji icons
 *
 */
public class EaseChatRowBigExpression extends EaseChatRowText {
    private ImageView imageView;

    public EaseChatRowBigExpression(Context context, boolean isSender) {
        super(context, isSender);
    }

    public EaseChatRowBigExpression(Context context, ChatMessage message, int position, BaseAdapter adapter) {
        super(context, message, position, adapter);
    }

    @Override
    protected void onInflateView() {
        inflater.inflate(!showSenderType ? R.layout.ease_row_received_bigexpression
                : R.layout.ease_row_sent_bigexpression, this);
    }

    @Override
    protected void onFindViewById() {
        percentageView = (TextView) findViewById(R.id.percentage);
        imageView = (ImageView) findViewById(R.id.image);
    }

    @Override
    public void onSetUpView() {
        String emojiconId = message.getStringAttribute(EaseConstant.MESSAGE_ATTR_EXPRESSION_ID, null);
        EaseEmojicon emojicon = null;
        if(EaseUIKit.getInstance().getEmojiconInfoProvider() != null){
            emojicon =  EaseUIKit.getInstance().getEmojiconInfoProvider().getEmojiconInfo(emojiconId);
        }
        if(emojicon != null){
            if(emojicon.getBigIcon() != 0){

                Glide.with(context).load(emojicon.getBigIcon())
                        .apply(RequestOptions.placeholderOf(R.drawable.ease_default_expression))
                        .into(imageView);
            }else if(emojicon.getBigIconPath() != null){
                Glide.with(context).load(emojicon.getBigIconPath())
                        .apply(RequestOptions.placeholderOf(R.drawable.ease_default_expression))
                        .into(imageView);
            }else{
                imageView.setImageResource(R.drawable.ease_default_expression);
            }
        }

        setImageIncludeThread(imageView);
    }
}

File: ease-im-kit/src/main/kotlin/com/hyphenate/easeui/widget/chatrow/EaseChatRowBigExpression.kt
package com.hyphenate.easeui.widget.chatrow

import android.content.Context
import android.util.AttributeSet
import android.widget.ImageView
import coil.load
import com.hyphenate.easeui.EaseIM
import com.hyphenate.easeui.R
import com.hyphenate.easeui.common.EaseConstant
import com.hyphenate.easeui.model.EaseEmojicon

/**
 * big emoji icons
 *
 */
open class EaseChatRowBigExpression @JvmOverloads constructor(
    private val context: Context,
    private val attrs: AttributeSet? = null,
    private val defStyleAttr: Int = 0,
    isSender: Boolean = false
) : EaseChatRowText(context, attrs, defStyleAttr, isSender) {
    private val imageView: ImageView? by lazy { findViewById(R.id.image) }


    override fun onInflateView() {
        inflater.inflate(
            if (!isSender) R.layout.ease_row_received_bigexpression else R.layout.ease_row_sent_bigexpression,
            this
        )
    }

    override fun onSetUpView() {
        message?.run {
            val emojiconId: String = getStringAttribute(EaseConstant.MESSAGE_ATTR_EXPRESSION_ID, null)
            var emojicon: EaseEmojicon? = null
            if (EaseIM.getEmojiconInfoProvider() != null) {
                emojicon = EaseIM.getEmojiconInfoProvider()?.getEmojiconInfo(emojiconId)
            }
            emojicon?.let {
                if (it.bigIcon != 0) {
                    imageView?.load(it.bigIcon) {
                        placeholder(R.drawable.ease_default_expression)
                        error(R.drawable.ease_default_expression)
                    }
                } else if (!it.bigIconPath.isNullOrEmpty()) {
                    imageView?.load(it.bigIconPath) {
                        placeholder(R.drawable.ease_default_expression)
                        error(R.drawable.ease_default_expression)
                    }
                } else {
                    imageView?.load(R.drawable.ease_default_expression)
                }
            }
            //setImageIncludeThread(imageView)
        }
    }
}